🔥 Remove Hypothesis plugin for now (#5691)
pydantic/pydantic: docs/integrations/hypothesis.md

[`typing`](https://docs.python.org/3/library/typing.html) and
[`typing_extensions`](https://pypi.org/project/typing-extensions/) modules by default.

> Pydantic v2.0 drops built-in support for Hypothesis and no more ships with the integrated Hypothesis plugin.

Reviewer: something about temporarily dropping it while we decide on the way forward, and a link to the issue I just created on annotated types.

[`typing`](https://docs.python.org/3/library/typing.html) and
[`typing_extensions`](https://pypi.org/project/typing-extensions/) modules by default.

Pydantic v2.0 drops built-in support for Hypothesis and no more ships with the integrated Hypothesis plugin.

> We are removing the Hypothesis plugin in favor of the mechanism discussed here  removed the plugin as we hope it shouldn’t be necessary in future. We hope to bring back support for Hypothesis to Pydantic via the mechanism discussed in [Way to communicate more information between libraries · Issue #37 · annotated-types/annotated-types](https://github.com/annotated-types/annotated-types/issues/37).

Reviewer: ```suggestion
We are temporarily removing the Hypothesis plugin in favor of another mechanism discussed here - [Way to communicate more information between libraries · Issue #37 · annotated-types/annotated-types](https://github.com/annotated-types/annotated-types/issues/37).
```

> It is also possible that Hypothesis plugin will be back as a separate library for Pydantic v2.0+ series (see [V2: hypothesis plugin rewrite · Issue #4682 · pydantic/pydantic](https://github.com/pydantic/pydantic/issues/4682)).

Reviewer: ```suggestion
It is also possible that Hypothesis plugin will be back in Pydantic v2.0+ (see [V2: hypothesis plugin rewrite · Issue #4682 · pydantic/pydantic](https://github.com/pydantic/pydantic/issues/4682)).
```
